R/imputation.R

```diff
@@ -26,7 +26,11 @@ setMethod("impute", "MSnSet",
               ## imputaton methods
               ##
               if (method == "knn") {
-                  .eset <- impute.knn(exprs(object), ...)$data
+                  imp_res <- impute.knn(exprs(object), ...)
+                  .eset <- imp_res$data
+                  if(! is.null(imp_res$rng.state)){
+                    .Random.seed <<- imp_res$rng.state
```

Comment:
Instead of using the `<<-` operator which could have some side-effects because it looks in the parent environments until it finds a variable with the given name (I know it is very unlikely but if a user writes a wrapper function around `impute` and defines a `.Random.seed` variable inside this function it would be a hard to find bug.) I would suggest:
 ```suggestion
                    assign(".Random.seed", imp_res$rng.state, envir=.GlobalEnv)
```

Reply:
Yes, great point, I forgot that `<<-` might update the wrong variable.